### PR TITLE
[tidy-html5] Update to 5.8.0, dynamic linkage, unique link lib name

### PR DESCRIPTION
**Pull request by dg0yt**
- #### What does your PR fix?
  Updates tidy-html5 to the latest stable version.
  Disables doc, removing the libxslt dependency.
  Enables building with dynamic linkage.
  Drop `_static` suffix from the static lib name for Windows, to match the dynamic import lib name. (Disruptive for existing users now, but actually necessary to allow consuming with variable linkage, and to match the pc file.)
  Fix debug postfix in pc file.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  all, no

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  yes. (NB: "Static and shared variants often should be renamed to a common scheme.")

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  yes.

### CODE REVIEW

**Comment by dg0yt** [comment]
I will update this PR when #24935 is merged.

File: ports/tidy-html5/disable-doc.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 8efec25..d2edac2 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -508,7 +508,7 @@ endif ()
+ # Create man pages
+ #################################################
+ 
+-if (UNIX AND SUPPORT_CONSOLE_APP)
++if (0)
```
Comment (LilyWangLL) [inline ports/tidy-html5/disable-doc.patch:10]: Could this patch instead by set `SUPPORT_CONSOLE_APP=OFF` in `vcpkg_cmake_configure`?

Comment (dg0yt) [reply ports/tidy-html5/disable-doc.patch:10]: I think that some users may actually want to use the `tidy` app. 
I only wanted to disable the installation of the documentation.

Comment (LilyWangLL) [reply ports/tidy-html5/disable-doc.patch:10]: Fine.

**Review by LilyWangLL — DISMISSED** [review]

**Review by FrankXie05 — DISMISSED** [review]

File: versions/t-/tidy-html5.json [hunk 2]
```diff
@@ -1,9 +1,9 @@
 {
   "versions": [
     {
-      "git-tree": "2e4c85fc988591dc5b25baac22c4e83dde7b7f64",
-      "version": "5.7.28",
-      "port-version": 6
+      "git-tree": "e16455cbec8214de37e3deb0ad41fe57323a7dd0",
+      "version": "5.8.0",
+      "port-version": 0
```
[the remark this answers is not in the document]
Comment (Thomas1664) [reply versions/t-/tidy-html5.json:6]: Maybe we should check the git diff in the Python script if there are only additions to this file.

**Pull request merged**